build/ArtifactBuilder/Artifacts/LinuxPackage.cs

```diff
@@ -9,35 +12,149 @@ public class LinuxPackage : Artifact
+                    var cpioFile = packageFileName.Replace(".rpm", "") + ".cpio.xz";
+                    using (ArchiveFile archiveFileXz = new ArchiveFile(Path.Join(unpackedDir, cpioFile)))
+                    {
+                        MemoryStream memoryStream = new MemoryStream();
+                        archiveFileXz.Entries[0].Extract(memoryStream);
+
+                        using (ArchiveFile archiveFileCpio = new ArchiveFile(memoryStream, SevenZipFormat.Cpio))
+                        {
+                            archiveFileCpio.Extract(unpackedDir);
+                        }
+                    }
+                    return Path.Combine(unpackedDir, "usr", "local", "newrelic-dotnet-agent");
+                case "tar.gz":
+                    // The SevenZipWrapper is choking on our tar.gz file for some reason
+                    Directory.CreateDirectory(unpackedDir);
+                    UnGzip(_packagePath, unpackedDir);
+                    using (ArchiveFile archiveFileTar = new ArchiveFile(Path.Join(unpackedDir, packageFileName.Replace(".gz", ""))))
+                    {
+                        archiveFileTar.Extract(unpackedDir);
+                    }
+                    return Path.Combine(unpackedDir, "newrelic-dotnet-agent");
+                default:
+                    throw new PackagingException($"Unknown extension {_fileExtension} for Linux package {_packageName}");
+            }
+        }
+
+        private void UnGzip(string gzipFile, string path)
+        {
+            using FileStream compressedFileStream = File.Open(gzipFile, FileMode.Open);
+            using FileStream outputFileStream = File.Create(Path.Join(path, new FileInfo(gzipFile).Name.Replace(".gz", "")));
+            using var decompressor = new GZipStream(compressedFileStream, CompressionMode.Decompress);
+            decompressor.CopyTo(outputFileStream);
+        }
+
+        private void ValidateContent()
+        {
+            var unpackedLocation = Unpack();
+
+            var installedFilesRoot = unpackedLocation;
+
+            var expectedComponents = GetExpectedComponents(installedFilesRoot);
+
+            var unpackedComponents = ValidationHelpers.GetUnpackedComponents(installedFilesRoot);
+
+            ValidationHelpers.ValidateComponents(expectedComponents, unpackedComponents, _packageName);
+
+            FileHelpers.DeleteDirectories(unpackedLocation);
         }
+
+        private SortedSet<string> GetExpectedComponents(string installedFilesRoot)
+        {
+            var expectedComponents = new SortedSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            ValidationHelpers.AddFilesToCollectionWithNewPath(expectedComponents, installedFilesRoot, _coreAgentComponents.RootInstallDirectoryComponents);
+            ValidationHelpers.AddFilesToCollectionWithNewPath(expectedComponents, installedFilesRoot, _coreAgentComponents.AgentHomeDirComponents.Where(f => !f.Contains("NewRelic.Profiler.dll")));
```

Comment:
You might be able to use something like `f != _coreAgentComponents.WindowsProfiler`.

Reply:
Yes, that works.